dispatcher/stateful.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+module.exports = function StatefulDispatcher(dispatcher, data, onData) {
```

Comment:
What about `StatelessDispatcher` and `StatefulDispatcher` and an additional doc string that explains that a `StatelessDispachter` can be used for stateless server rendering and the `StatefulDispatcher` is useful for a stateful client?